### DISCUSSION TITLE
If playerSelector is removed from DOM…

### DISCUSSION THREAD

**Discussion by hornbuckle** [web-scrobbler/web-scrobbler#3214] [Q&A]
…can the connector be reset (without manually refreshing the browser window) to detect it again if/when it gets re-added?

**Comment by inverse** [comment]
Perhaps this is what you're after?

https://github.com/web-scrobbler/web-scrobbler/blob/master/src/connectors/focusatwill.js

This gets called when the connector is injected.

**Reply by hornbuckle** [reply]
@inverse Sort of, but maybe not quite. The MutationObserver would need to observe the playerSelector. If it is removed from the DOM (when an album stops playing) and later re-added to the DOM (when the album is played again or a new album is selected and played), what function would reinitialize the scrobbler, or is that even available?

**Reply by inverse** [reply]
Hmmm - in such a situation wouldn't we just need to invoke `Connector.onStateChanged` to trigger the state update?

It's been a while since I dived into the core :D

**Reply by hornbuckle** [reply]
@inverse Okay, thanks, I'll try it out.

**Reply by hornbuckle** [reply]
@inverse I may have been overthinking what needed to be done here. I tried writing a MutationObserver, but I couldn't get it quite right. I could detect when the player class was removed, but invoking onStateChanged didn't seem to work as expected, as it still couldn't find the player class when it was re-added to the DOM.
Then I noticed this comment on playerSelector in connector.js: "Selector of a container _closest to_ the player."
I always thought the playerSelector should be the outermost wrapper of the player itself. However, when I update playerSelector to a parent element (`#app`—it's a React site) instead of the player wrapper itself, everything still functions as expected _and_ the connector still finds everything again after it's destroyed and added to the page again when replaying the album because playerSelector includes its own observer.
Considering that this solves the issue/question, would this be an acceptable update to submit for the connector?

**Reply by inverse** [reply]
@hornbuckle the `playerSelector` is used to setting up the mutation stuff for watching for state changes.

I think we should aim to target things as best we can but it sounds like what you described is a good workaround for the site in question.
